packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx

```diff
@@ -49,124 +50,109 @@ export interface State {
   isOpen: boolean;
 }
 
-export class UnthemedTimeRangePicker extends PureComponent<TimeRangePickerProps, State> {
-  state: State = {
-    isOpen: false,
+export const UnthemedTimeRangePicker = (props: TimeRangePickerProps) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const {
+    value,
+    onMoveBackward,
+    onMoveForward,
+    onZoom,
+    timeZone,
+    fiscalYearStartMonth,
+    timeSyncButton,
+    isSynced,
+    theme,
+    history,
+    onChangeTimeZone,
+    onChangeFiscalYearStartMonth,
+    hideQuickRanges,
+  } = props;
+
+  const onChange = (timeRange: TimeRange) => {
+    props.onChange(timeRange);
+    setOpen(false);
   };
 
-  onChange = (timeRange: TimeRange) => {
-    this.props.onChange(timeRange);
-    this.setState({ isOpen: false });
-  };
-
-  onOpen = (event: FormEvent<HTMLButtonElement>) => {
-    const { isOpen } = this.state;
+  const onOpen = (event: FormEvent<HTMLButtonElement>) => {
     event.stopPropagation();
     event.preventDefault();
-    this.setState({ isOpen: !isOpen });
+    setOpen(!isOpen);
   };
 
-  componentDidMount() {
-    window.addEventListener('keyup', this.onKeyUp);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('keyup', this.onKeyUp);
-  }
-
-  onKeyUp = (event: KeyboardEvent) => {
-    if (event.code === 'Escape') {
-      this.onClose();
-    }
+  const onClose = () => {
+    setOpen(false);
   };
 
-  onClose = () => {
-    this.setState({ isOpen: false });
-  };
+  const ref = createRef<HTMLElement>();
+  const { overlayProps } = useOverlay({}, ref);
```

Comment:
you just need to pass the relevant props here, then everything should work:

```suggestion
  const { overlayProps } = useOverlay({ isOpen, onClose }, ref);
```

see https://react-spectrum.adobe.com/react-aria/useOverlay.html and click the `OverlayProps` type